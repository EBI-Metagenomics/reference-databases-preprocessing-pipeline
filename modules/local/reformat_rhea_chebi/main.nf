process REFORMAT_RHEA_CHEBI {
    label 'process_single'
    container 'quay.io/biocontainers/python:3.10'

    input:
    path txt_rhea_chebi_mapping 

    output:
    path 'rhea_chebi_mapping_*.tsv', emit: tsv_rhea_chebi_mapping

    script:
    
    """
    reformat_rhea_chebi_mapping.py ${txt_rhea_chebi_mapping} rhea_chebi_mapping_${params.rheadb_version}.tsv
    """
}