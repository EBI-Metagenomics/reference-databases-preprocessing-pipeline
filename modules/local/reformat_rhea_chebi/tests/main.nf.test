nextflow_process {

    name "Test Process REFORMAT_RHEA_CHEBI"
    script "../main.nf"
    process "REFORMAT_RHEA_CHEBI"

    test("Convert test file to TSV format") {

        when {
            params {
                rheadb_version = "135"
            }
            process {
                """
                input[0] = Channel.fromPath("${moduleDir}/tests/data/rhea-reactions.txt.gz", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
