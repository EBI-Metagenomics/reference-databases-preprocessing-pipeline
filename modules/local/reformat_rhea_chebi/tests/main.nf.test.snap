{
    "Convert test file to TSV format": {
        "content": [
            {
                "0": [
                    "rhea_chebi_mapping_135.tsv:md5,52d55d130ac9e9329e6e32d88b335f58"
                ],
                "1": [
                    "versions.yml:md5,1fe56da75cf247da6549287a99658476"
                ],
                "tsv_rhea_chebi_mapping": [
                    "rhea_chebi_mapping_135.tsv:md5,52d55d130ac9e9329e6e32d88b335f58"
                ],
                "versions": [
                    "versions.yml:md5,1fe56da75cf247da6549287a99658476"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-19T15:50:00.476298"
    }
}